src/table/utils/styling-utils.ts

```diff
@@ -40,34 +51,49 @@ export function getPadding(styleObj, defaultPadding) {
   return padding;
 }
 
-export function getColor(defaultColor, theme, color = {}) {
+/**
+ * Gets color from color picker. Defaults to default color if resolved color is invalid
+ */
+export function getColor(defaultColor: string, theme: stardust.Theme, color = {}): string {
   const resolvedColor = theme.getColorPickerColor(color);
 
   return !resolvedColor || resolvedColor === 'none' ? defaultColor : resolvedColor;
 }
 
-export const getAutoFontColor = (backgroundColor) =>
+/**
+ * Gets font color based on background, making sure contrast is good enough
+ */
+export const getAutoFontColor = (backgroundColor: string): string =>
   isDarkColor(backgroundColor) ? STYLING_DEFAULTS.WHITE : STYLING_DEFAULTS.FONT_COLOR;
 
-export const getBaseStyling = (styleObj, objetName, theme) => {
+/**
+ * Gets base styling for either header or body taking table theme settings into account
+ */
+export const getBaseStyling = (styleObj: StylingLayout, objetName: string, theme: ExtendedTheme): GeneratedStyling => {
   const fontFamily = theme.getStyle('object', `straightTable.${objetName}`, 'fontFamily');
   const color = theme.getStyle('object', `straightTable.${objetName}`, 'color');
   const fontSize = theme.getStyle('object', `straightTable.${objetName}`, 'fontSize');
 
-  const baseStyle = {
+  const baseStyle: GeneratedStyling = {
     fontFamily,
-    color: isSet(styleObj?.fontColor) ? getColor(STYLING_DEFAULTS.FONT_COLOR, theme, styleObj.fontColor) : color,
-    fontSize: styleObj?.fontSize || fontSize,
+    color: isColorSet(styleObj?.fontColor) ? getColor(STYLING_DEFAULTS.FONT_COLOR, theme, styleObj.fontColor) : color,
+    fontSize: (styleObj?.fontSize && `${styleObj.fontSize}px`) || fontSize,
```

Comment:
Does `theme.getStyle('object', `straightTable.${objetName}`, 'fontSize');` contain 'px' as well?

Reply:
yes it should be

Reply:
ok, good